pkg/eventcollector/eventcollector.go

```diff
@@ -140,6 +145,13 @@ func (c *EventCollector) handleEvent(payload interface{}) interface{} {
 	}
 	log.Infof("handleEvent: events saved") // Debug, remove later
 
+	if c.crawlerPubSub != nil {
```

Comment:
Any reason why this code is here and not after everything else is done?

Reply:
I just wanted to send it right after we saved it to the DB. If it is an application event it goes through that whole logic of filtering past events as well, and didn't want to wait for that. That reminds me though that we should probably send those newsroom events to the processor as well as watched messages, then query for that eventHash to process that newsroom event (rather than query by timestamp)